lib/compress.js

```diff
@@ -183,9 +184,15 @@ merge(Compressor.prototype, {
                 node.walk(new TreeWalker(function() {
                     count++;
                 }));
-                this.info("pass " + pass + ": last_count: " + last_count + ", count: " + count);
-                if (count >= last_count) break;
-                last_count = count;
+                this.info("pass " + pass + ": last_count: " + min_count + ", count: " + count);
+                if (count < min_count) {
+                    min_count = count;
+                    stopping = false;
```

Comment:
Won't this increase the number of passes on average? At least the hard limit of `passes` is still respected as far as I can tell.